Sources/ComposableArchitecture/SwiftUI/Alert.swift

```diff
@@ -128,47 +128,86 @@ public struct AlertState<Action> {
   }
 
   public struct Button {
-    public var action: Action?
-    public var animation: Animation??
-    public var type: `Type`
+    public var action: ButtonAction?
+    public var type: ButtonType
 
     public static func cancel(
       _ label: TextState,
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .cancel(label: label))
+      Self(action: action, type: .cancel(label: label))
     }
 
     public static func cancel(
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .cancel(label: nil))
+      Self(action: action, type: .cancel(label: nil))
     }
 
     public static func `default`(
       _ label: TextState,
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .default(label: label))
+      Self(action: action, type: .default(label: label))
     }
 
     public static func destructive(
       _ label: TextState,
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .destructive(label: label))
+      Self(action: action, type: .destructive(label: label))
     }
 
-    public enum `Type` {
-      case cancel(label: TextState?)
-      case `default`(label: TextState)
-      case destructive(label: TextState)
+    public static func cancel(
+      _ label: TextState,
+      send action: Action?
+    ) -> Self {
+      Self(action: action.map(ButtonAction.send), type: .cancel(label: label))
+    }
+
+    public static func cancel(
+      send action: Action?
+    ) -> Self {
+      Self(action: action.map(ButtonAction.send), type: .cancel(label: nil))
+    }
+
+    public static func `default`(
+      _ label: TextState,
+      send action: Action?
+    ) -> Self {
+      Self(action: action.map(ButtonAction.send), type: .default(label: label))
+    }
+
+    public static func destructive(
+      _ label: TextState,
+      send action: Action?
+    ) -> Self {
+      Self(action: action.map(ButtonAction.send), type: .destructive(label: label))
     }
   }
+
+  public struct ButtonAction {
+    let type: ActionType
+
+    public static func send(_ action: Action) -> Self {
+      .init(type: .send(action))
+    }
+
+    public static func send(_ action: Action, animation: Animation?) -> Self {
+      .init(type: .animatedSend(action, animation: animation))
+    }
+
+    enum ActionType {
+      case send(Action)
+      case animatedSend(Action, animation: Animation?)
+    }
```

Comment:
Using an internal enum here to work around https://bugs.swift.org/browse/SR-10077, and so we can exhaustively interpret things...but should it be `public` so people can write their own code that plucks out the values inside?

Reply:
(Also, this enum leaves us open to supporting UIKit animations in the future, should someone submit a PR.)

Reply:
@stephencelis re: supporting UIKit animations, would calling something like this in #563 potentially work?

``` swift
import struct SwiftUI.Animation
func withUIKitAnimation(_ animation: Animation?, body: @escaping () -> Void) -> Void {
  switch animation {
  case .default?:
    UIView.animate(
      withDuration: 0.35,
      delay: 0,
      options: [.curveEaseInOut, .allowUserInteraction],
      animations: body
    )
  ...
  default:
    body()
  }
}
```

Reply:
@andreyz I think we would add brand new enum cases that takes the arguments sent to `UIView.animate`. That would seem easier to get right than trying to convert a SwiftUI animation over to UIKit.

Reply:
@stephencelis do you mean hypothetically an extra `.animatedSend` -like case with explicit parameters for duration, options, etc? If so, I'll do it, after my earlier PR lands.

Reply:
I think so...the only complication is that animated Combine schedulers can at least nest, so you could perform UIKit and SwiftUI actions on the same action. It may sound unlikely, but we do just this in isowords for screens that show both UIKit and SwiftUI...probably less likely for alerts, though, so definitely feel free to start exploring a solution there!

Reply:
Thanks for encouragement and feedback @stephencelis. I'll check isowords and noodle on it.